app/controllers/buyers/accounts/applications_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class Buyers::Accounts::ApplicationsController < Buyers::BaseController
+
+  # include ThreeScale::Search::Helpers
+  # include DisplayViewPortion
+  # helper DisplayViewPortion::Helper
+
+  # before_action :authorize_partners
+  before_action :find_buyer, only: %i[index create new]
+  # before_action :find_plans, only: %i[new create]
+
+  # before_action :find_cinstance, :except => [:index, :create, :new]
+  # before_action :find_provider,  only: %i[new create update]
+
+  activate_menu :buyers, :accounts, :listing
+
+  def index
+    # TODO: Editing this action may require editing Api::ApplicationsController
+
+    @states = Cinstance.allowed_states.collect(&:to_s).sort
+    @search = ThreeScale::Search.new(params[:search] || params)
+    @application_plans = accessible_plans
+
+    if params[:service_id]
+      @service = accessible_services.find params[:service_id]
+      @search.service_id = @service.id
+    end
+
+    if params[:application_plan_id]
+      @plan = @application_plans.find params[:application_plan_id]
+      @search.plan_id = @plan.id
+      @service ||= @plan.service
+    end
+
+    @search.account = buyer.id
+
+    @cinstances = accessible_not_bought_cinstances.scope_search(@search)
+                                                  .order(params[:sort] => params[:direction])
+                                                  .preload(:service, user_account: [:admin_user], plan: [:pricing_rules])
+                                                  .paginate(pagination_params)
+                                                  .decorate
+
+    display_view_portion!(:service) if current_account.multiservice?
+  end
+
+  # This will be used with
+  # /buyers/accounts/3/applications/new
```

Comment:
2 comments here:
1. "buyers" and "accounts" literally mean the same thing in this context, so `/buyers/accounts/{id}` feels a bit weird. Could we have either just `buyers/{id}` or (better) `accounts/{id}`?
2. We have to decide if the Admin Portal controllers will be all under `app/controllers/provider/admin` or `app/controllers/apiconfig`, one of these, one of these and other, or pretty much anywhere under `app/controllers`